pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -142,50 +139,26 @@ func (r *PullSecretReconciler) pullsecret(ctx context.Context) (*corev1.Secret,
 	return ps, false, nil
 }
 
-func triggerReconcile(secret *corev1.Secret) bool {
-	return (secret.Name == pullSecretName.Name && secret.Namespace == pullSecretName.Namespace) ||
-		(secret.Name == operator.SecretName && secret.Namespace == operator.Namespace)
-}
-
 // SetupWithManager setup our manager
 func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	isPullSecret := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			_, ok := e.ObjectOld.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
-
-			secret, ok := e.ObjectOld.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-		CreateFunc: func(e event.CreateEvent) bool {
-			_, ok := e.Object.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
-
-			secret, ok := e.Object.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-		DeleteFunc: func(e event.DeleteEvent) bool {
-			_, ok := e.Object.(*arov1alpha1.Cluster)
-			if ok {
-				return true
-			}
+	pullSecretPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+		return (meta.GetName() == pullSecretName.Name && meta.GetNamespace() == pullSecretName.Namespace) ||
+			(meta.GetName() == operator.SecretName && meta.GetNamespace() == operator.Namespace)
+	})
 
-			secret, ok := e.Object.(*corev1.Secret)
-			return ok && triggerReconcile(secret)
-		},
-	}
+	aroClusterPredicate := predicate.NewPredicateFuncs(func(meta metav1.Object, object runtime.Object) bool {
+		return meta.GetName() == arov1alpha1.SingletonClusterName
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&arov1alpha1.Cluster{}).
+		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
 		// https://github.com/kubernetes-sigs/controller-runtime/issues/1173
 		// equivalent to For(&v1.Secret{})., but can't call For multiple times on one builder
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}).
-		Owns(&corev1.Secret{}).
```

Comment:
Removing `Owns` because: 

1. This reconciler doesn't set controller references,  but this `Owns` was relying on the reference set by `GenevaloggingReconciler` which reconciles `operator.SecretName` - it is a bit confusing.

2. `Watches` coevers both  `pullSecretName.Name` and `operator.SecretName` secrerts.

3. `EnqueueRequestForObject` or `EnqueueRequestForOwner` handlers are equally ok for this reconciler as we get both secrets by hardcoded names.

Reply:
Oh, actually I think I found a potential issue: in previous comment I said that `GenevaloggingReconciler` sets controller references to `operator.SecretName`. It is not the case. It is not managing `operator.SecretName`. It is being deployed here:

https://github.com/Azure/ARO-RP/blob/7cc1a512534de08cb356527c4907d03111b593c0/pkg/operator/deploy/deploy.go#L156-L166

And it is not reconciled.

So `oc -n openshift-azure-operator delete secret cluster` breaks things. Together with `oc -n openshift-config delete secret pull-secret` breaks pull secret completely. Can only be recovered by SRE admin update.